### PR TITLE
Added login functionality

### PR DESCRIPTION
**Pull request by cahya-wirawan**
# Description

Currently before we are able to use the udacimak, we have to login to udacity as usual using 
a web browser and get the _jwt token from the developer tool and use it in udacimak. To simplify 
and automate this process, I have created the login functionality. What we have to do is just exec
'udacimak login', it will prompt for the email address and password of udacity account. If the
authentication is successful, it will use the function validateSaveUdacityAuthToken to save the
retrieved token (_jwt). Therefore 'udacimak settoken YOUR_UDACITY_AUTH_TOKEN' is not needed anymore.
Btw, this is my first PR to a nodejs project, so if I don't follow the style guideline,  then just let me know, I will try to fix it.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
cahya@arjuna: $ npm link
cahya@arjuna: $ rm -f ~/.udacimak
cahya@arjuna: $ udacimak login
prompt: email:  test@example.com
prompt: password:  
Invalid login credentials
cahya@arjuna: $ udacimak login
prompt: email:  cahya@example.com
prompt: password:  
✔ Validate Udacity authentication token via Udacity API
2020-03-31T15:44:12.781Z [info]: Hi cahya@example.com. You have successfully saved authentication token locally!
cahya@arjuna: $ udacimak listnd
✔ Fetch user information via Udacity API
2020-03-31T15:44:39.906Z [info]: Hi cahya@example.com!
2020-03-31T15:44:39.907Z [info]: 
2020-03-31T15:44:39.907Z [info]: Here's a list of your enrolled Nanodegrees:
2020-03-31T15:44:39.908Z [info]:  - nd013 version 3.0.0 (locale: en-us): Self-Driving Car Engineer
2020-03-31T15:44:39.908Z [info]:  - nd209 version 3.0.0 (locale: en-us): Robotics Software Engineer
2020-03-31T15:44:39.908Z [info]:  - nd009-ent version 4.0.0 (locale: en-us): Machine Learning Engineer

# Configurations
 - OS: Ubuntu 18.04
 - Node version: v12.16.1
 - npm version: 6.13.7

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
     "regenerator-runtime": "^0.12.1"
   },
   "dependencies": {
+    "axios": "^0.19.2",
```
Comment (udacimak) [inline package.json:69]: Please use `request` to implement API requests instead as we don't need to use a third-party library. 

Refer to some of the code in `src/api` directory

File: src/commands/functions/authenticate.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import {
+  validateSaveUdacityAuthToken,
+} from '.';
+
+const axios = require('axios');
+const prompt = require('prompt');
```
Comment (udacimak) [inline src/commands/functions/authenticate.js:6]: We're already using `inquirer` library for prompts, please update code to use `inquirer` instead. 

You can look at `promptInputUdacityAuthToken.js`  or `promptOverwriteDir.js` file as an example

**Comment by cahya-wirawan** [comment]
Hi, axios and prompt have been replaced with request and inquirer. Btw, the module request is deprecated: https://github.com/request/request/issues/3142, in case you plan to update it.

**Comment by udacimak** [comment]
> Hi, axios and prompt have been replaced with request and inquirer. Btw, the module request is deprecated: [request/request#3142](https://github.com/request/request/issues/3142), in case you plan to update it.

Thanks! I'll leave `request` module there for now and see what happens in the future

**Pull request merged**